sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ServiceUnavailableException.java

```diff
@@ -79,5 +86,12 @@ public ServiceUnavailableException(String message,
             HttpUtils.asMap(headers),
             HttpConstants.StatusCodes.SERVICE_UNAVAILABLE,
             requestUriString);
+        setSubStatus(subStatusCode);
+    }
+
+    private void setSubStatus(int subStatusCode) {
+        this.getResponseHeaders().put(
```

Comment:
I guess we can also use `BridgeInternal.setSubStatusCode(this, subStatusCode);`

Reply:
Discussed offline, going to have a followup PR to refactor it as there are multiple duplicate places